core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -271,6 +271,15 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
     rdd.pipe(command.asScala, env.asScala)
 
   /**
+   * Return an RDD created by piping elements to a forked external process.
+   */
+  def pipe(command: JList[String],
+           env: java.util.Map[String, String],
```

Comment:
Optional: wouldn't mind importing this as JMap and using it consistently in this file while we're here.

This can also take arguments for `printPipeContext` and `printRDDElement`. That's not part of your change of course but make sense to fix this omission while adding the all-args API method to support the new arg.